utils/src/modules/repository.rs

```diff
@@ -32,19 +36,43 @@ impl Repository {
     }
 
     pub fn update_at(&mut self, key: String, value: Bytes, activation_time: Option<u64>) {
-        let mode = Repository::resolve_update_mode(activation_time);
-        let (current, _) = self.storage.get(&key);
-        let new_value = match mode {
-            UpdateMode::SetFuture => (current, Some((value.clone(), activation_time.unwrap()))),
-            UpdateMode::ClearFuture => (current, None),
-            UpdateMode::Current => (value.clone(), None),
+        let now = get_block_time();
```

Comment:
`get()` method should be refactored as well to approach the block time the same way

Reply:
done